libsolidity/codegen/ir/IRGenerationContext.h

```diff
@@ -98,10 +98,10 @@ class IRGenerationContext
 
 	void addStateVariable(VariableDeclaration const& _varDecl, u256 _storageOffset, unsigned _byteOffset);
 	bool isStateVariable(VariableDeclaration const& _varDecl) const { return m_stateVariables.count(&_varDecl); }
-	std::pair<u256, unsigned> storageLocationOfVariable(VariableDeclaration const& _varDecl) const
-	{
-		return m_stateVariables.at(&_varDecl);
-	}
+	std::pair<u256, unsigned> storageLocationOfStateVariable(VariableDeclaration const& _varDecl) const;
+
+	std::pair<std::string, std::string> storageLocationOfLocalVariable(VariableDeclaration const& _varDecl) const;
+	std::pair<std::string, std::string> storageLocationOfVariable(VariableDeclaration const& _varDecl) const;
```

Comment:
I'm not sure I understand the different signatures here i.e., `u256, unsigned` vs `string, string`

Reply:
StateVariables have numeric literals for `slot/offset` and LocalVariables have parts for `slot/offset` which are `IRVariables`.

There is an option to return string in both cases and have only one function. Does it sound better than?

Reply:
To me, inlining `storageLocationOfLocalVariable` into `storageLocationOfVariable` and having the latter return `string, string` and `storageLocationOfStateVariable` return `u256, unsigned` seems reasonable